addons/website/controllers/form.py

```diff
@@ -259,7 +259,7 @@ def insert_attachment(self, model, id_record, files):
             }
             attachment_id = request.env['ir.attachment'].sudo().create(attachment_value)
             if attachment_id and not custom_field:
-                record.sudo()[file.field_name] = [(4, attachment_id.id)]
+                record.sudo()[file.field_name] = attachment_id.id
```

Comment:
This should probably actually be the `[(4, id)]` notation for x2many fields and the direct id assignation for m2o, no?
Because this will crash for x2many (the same error you are tying to fix -> wrong value), on top of not adding the attachment alongside the existing ones for that field but replacing all of theme by the new one.

Reply:
Most likely the other use case exists as well, so I guess you are right 👍🏼